http.go

```diff
@@ -16,23 +19,40 @@ const (
 	chunkThreshold = 5 * 1000000 // 5 megabytes
 )
 
+var httpChunkBufPool = sync.Pool{
```

Comment:
This is a performance optimisation and _hugely_ reduces memory allocations. When running `BenchmarkRequestResponse`, it has shaved off >600MB of allocations.